students/Briadean/lesson03/mailroom_pt1-2.py

```diff
@@ -0,0 +1,149 @@
+        print(donor)
+        # Why is this printing None at the end of the list?
+
+
+def accept_donation(name):
+    """Accept a new donation. If donor does not already exist, add to dictionary
+    and create appropriate data containerself.
+    Params: input = donor name
+            returns a thank you letter
+    """
+    donors_dict.setdefault(name, [])
+    donation_amt = input("How much are they donating today? >> $ ")
+    donors_dict[name].append(float(donation_amt))
+    print(generate_thank_you(name, float(donation_amt)))
+
+
+def generate_thank_you(name, donation):
+    """Generates a standardized thank you letter
+    Params: input = donor name, most recent donation amounts
+            returns a write_letters_disk
+    """
+    letter = ("\nDear {},\n\n"
+              "Thank you for your recent generous donation of ${:,.2f}.\n"
+              "It will be put to good use.\n\n"
+              "Sincerely,\n"
+              "The Resistance Against Giant Flying Meatballs\n".format(name, donation))
+    return letter
+
+
+def write_letters_disk():
+    """Writes a thank you letter for each donor in the database to disk"""
+    for donor, donation in donors_dict.items():
+        letter = generate_thank_you(donor, donation[-1])
+        filename = donor + ".txt"
+        print("Writing letter for:", donor)
+        open(filename, "w").write(letter)
+
+
+def sort_key(item):
+    """Sort function
+    Params: input = "item" or "entry" in the dictionary
+            returns the "Total Given" column entry
+    """
+    return item[1]
+
+
+def generate_report():
+    """Build a report of donors"""
+    report_lines = []
+    for (name, gifts) in donors_dict.items():
+        total_given = sum(gifts)
+        num_gifts = len(gifts)
+        average_gifts = total_given / num_gifts
+        report_lines.append((name, total_given, num_gifts, average_gifts))
+
+    report_header = "\n{:20} | {:12} | {:11} | {:12}".format("Donor Name",
+                                                             "Total Given",
+                                                             "Num Gifts",
+                                                             "Average Gifts"
+                                                             )
+
+    report_lines.sort(key=sort_key, reverse=True)
+    report = []
+    report.append(report_header)
+    report.append("-" * 55)
+    for line in report_lines:
+        report.append("{:20}  ${:12,.2f}  {:11}  ${:12,.2f}".format(* line))
+
+    return "\n".join(report)
+
+
+def display_report():
+    """Display the generated report"""
+    print(generate_report())
+
+
+def exit_program():
+    response = input("\nAre you sure you'd like to exit the program? Y/N>> ").capitalize()
+    valid_response = ["Y", "N"]
+    while response not in valid_response:
+        response = input("{} is not an available option. Please enter Y or N>> ".format(response))
+    while response in valid_response:
+        if response == "N":
```

Comment:
So an option, if you want, is to say if response[0] == "N" then if a user types "no" or "yes" it won't throw an error. Of course then the user can also type "yaldkja" and it'd think that's a "yes." Either way works